prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -5496,10 +5504,41 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                 let (expr, mut stmts) = self.postprocess_place_encoding(base)?;
                 (expr.field(field), stmts)
             }
-            PlaceEncoding::ArrayAccess { base, index, array_elem_ty, array_len } => {
-                return Err(EncodingError::unsupported(
-                    "array operations are not supported yet"
-                ));
+            PlaceEncoding::ArrayAccess { base, index, array_elem_ty, array_len, lookup_pure_ret, val_field } => {
+                let elem_ty_name = if let vir::Type::TypedRef(ref name) = array_elem_ty { name } else { unreachable!() };
+                let lookup_pure = self.encoder.encode_builtin_function_use(
+                    BuiltinFunctionKind::ArrayLookupPure {
+                        array_elem_ty: array_elem_ty.clone(),
+                        array_len,
+                        return_ty: lookup_pure_ret.clone(),
+                    }
+                );
+
+                let array_type = base.get_type().clone();
+                let lookup_res: vir::Expr = self.cfg_method.add_fresh_local_var(array_elem_ty).into();
+                let lookup_res_val_int = lookup_res.clone().field(val_field.clone());
+                let (encoded_base_expr, mut stmts) = self.postprocess_place_encoding(*base)?;
+                stmts.extend(self.encode_havoc_and_allocation(&lookup_res));
+                let lookup_pure_call = vir::Expr::func_app(
+                    lookup_pure,
+                    vec![encoded_base_expr, index.field(vir::Field::new("val_int", vir::Type::Int))],
```

Comment:
In case you need to build thinks like `index.field(vir::Field::new("val_int", vir::Type::Int))` again, `Encoder::encode_value_expr` might be slightly better. (No need to change this PR, snapshots are going to change this.)